### PR TITLE
feat: add typedef for useLocation hook

### PR DESCRIPTION
**Pull request by HenryKenya**
- [react-router_v1.1.x]: add type defintion for useLocation hook
- Link to documentation — [https://reach.tech/router/api/useLocation](https://reach.tech/router/api/useLocation)

### CODE REVIEW

**Comment by HenryKenya** [comment]
@GAntoine opened a PR regarding issue [#3916](https://github.com/flow-typed/flow-typed/issues/3916)

File: definitions/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,2627 @@
+{
```
Comment (GAntoine) [inline definitions/package-lock.json:1]: Remove this file, we use yarn for our lock files.

**Review by GAntoine — CHANGES_REQUESTED** [review]
Looks like the specs due to code that wasn't added here. If you can fix the specs that would be great, but if not, I'm happy to merge this in once the lock file's been removed.

File: definitions/package-lock.json (hunk 1 above)
Comment (HenryKenya) [reply definitions/package-lock.json:1]: Let me go ahead and do that

Comment (HenryKenya) [reply definitions/package-lock.json]: @GAntoine removed the package-lock file and updated the comment to reflect the affected spec.

File: definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/router_v1.1.x.js [hunk 2]
```diff
@@ -130,4 +137,6 @@ declare module '@reach/router' {
   declare export function isRedirect(error: any): $Exact<{ uri: string }>;
 
   declare export function redirectTo(uri: string): void;
+
+  declare export function useLocation(): $PropertyType<UseLocation, 'location'>;
```
Comment (pascalduez) [inline definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/router_v1.1.x.js:141]: Well, there's no `location` key in the `UseLocation` type you added.

Comment (HenryKenya) [reply definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/router_v1.1.x.js:141]: @pascalduez resolved with current [pull request](https://github.com/flow-typed/flow-typed/pull/3937/)

**Pull request merged**